lightning/api_clnsocket_test.go

```diff
@@ -237,3 +239,75 @@ func TestClnGetChanInfo(t *testing.T) {
 		t.Fatal("Wrong response")
 	}
 }
+
+type RawMethodCall func(ctx context.Context, api LightingAPICalls) ([]RawMessage, error)
+
+func rawCommon(t *testing.T, file string, method string, call RawMethodCall) {
+
+	data := clnData(t, file)
+
+	_, api, closer := clnCommon(t, func(c net.Conn) {
+		buf := make([]byte, BUFSIZE)
+		n, err := c.Read(buf)
+		if err != nil {
+			t.Fatalf("Could not read request body: %v", err)
+		}
+
+		// Reslice else the thing contains zero bytes
+		buf = buf[:n]
+		s := string(buf)
+
+		id := IDExtractor{}
+		err = json.Unmarshal(buf, &id)
+		if err != nil {
+			t.Fatalf("Unmarshal error: %v", err)
+		}
+
+		if strings.Contains(s, method) {
+			reply := fmt.Sprintf(string(data), id.ID)
+			_, err = c.Write(([]byte)(reply))
+
+			if err != nil {
+				t.Fatalf("Could not write to socket: %v", err)
+			}
+		}
+
+		err = c.Close()
+		if err != nil {
+			t.Fatalf("Could not close socket: %v", err)
+		}
+	})
+	defer closer()
+
+	resp, err := call(context.Background(), api)
+	assert.NoError(t, err)
+	assert.Equal(t, 10, len(resp))
+	assert.Equal(t, 2023, resp[0].Timestamp.Year())
+}
+
+func TestClnGetForwardsRaw(t *testing.T) {
+	rawCommon(t, "cln_listforwards", "listforwards",
+		RawMethodCall(func(ctx context.Context, api LightingAPICalls) ([]RawMessage, error) {
```

Comment:
would make sense to change RawMethodCall to have same api to avoid even more duplication